core/src/test/java/com/linecorp/armeria/client/HttpClientExpect100HeaderTest.java

```diff
@@ -0,0 +1,492 @@
+                    final OutputStream out = s.getOutputStream();
+                    assertThat(in.readLine()).isEqualTo("POST / HTTP/1.1");
+                    assertThat(in.readLine()).startsWith("host: 127.0.0.1:");
+                    assertThat(in.readLine()).isEqualTo("content-type: text/plain; charset=utf-8");
+                    assertThat(in.readLine()).isEqualTo("expect: 100-continue");
+                    assertThat(in.readLine()).isEqualTo("content-length: 3");
+                    assertThat(in.readLine()).startsWith("user-agent: armeria/");
+                    assertThat(in.readLine()).isEmpty();
+
+                    out.write(("HTTP/1.1 417 Expectation Failed\r\n" +
+                               "Connection: close\r\n" +
+                               "\r\n").getBytes(StandardCharsets.US_ASCII));
+
+                    assertThat(in.readLine()).isNull();
+                }
+            }
+        }
+
+        @Test
+        void continueToSendHttp2Request() throws Exception {
+            try (ServerSocket ss = new ServerSocket(0);
+                 ClientFactory clientFactory =
+                         ClientFactory.builder()
+                                      .useHttp2Preface(true)
+                                      .http2InitialConnectionWindowSize(Http2CodecUtil.DEFAULT_WINDOW_SIZE)
+                                      .http2InitialStreamWindowSize(Http2CodecUtil.DEFAULT_WINDOW_SIZE)
+                                      .build()) {
+                final int port = ss.getLocalPort();
+                final WebClient client = WebClient.builder("http://127.0.0.1:" + port)
+                                                  .factory(clientFactory)
+                                                  .build();
+                final CompletableFuture<AggregatedHttpResponse> future =
+                        client.prepare()
+                              .post("/")
+                              .content("foo")
+                              .header(HttpHeaderNames.EXPECT, HttpHeaderValues.CONTINUE)
+                              .execute()
+                              .aggregate();
+
+                try (Socket s = ss.accept()) {
+                    final InputStream in = s.getInputStream();
+                    final BufferedOutputStream bos = new BufferedOutputStream(s.getOutputStream());
+
+                    // Read the connection preface and discard it.
+                    readBytes(in, connectionPrefaceBuf().capacity());
+
+                    // Read a SETTINGS frame and validate it.
+                    readSettingsFrame(in);
+                    sendEmptySettingsAndAckFrame(bos);
+
+                    readBytes(in, 9); // Read a SETTINGS_ACK frame and discard it.
+
+                    // Read a HEADERS frame and validate it.
+                    readHeadersFrame(in);
+                    // Send a CONTINUE response.
+                    sendFrameHeaders(bos, HttpStatus.CONTINUE, false);
```

Comment:
Ditto - we need to ensure the client doesn't send anything between these two steps.